src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -95,6 +95,81 @@
     }
 
     /**
+     * Generates the email containing the summary of the feedback sessions
+     * email for the given {@code courseId} for {@code student}
+     * @param courseId - ID of the course
+     * @param student - attributes of student to send feedback session summary to
+     */
+    public EmailWrapper generateFeedbackSessionSummaryOfCourse(String courseId, StudentAttributes student) {
+
+        CourseAttributes course = coursesLogic.getCourse(courseId);
+
+        List<FeedbackSessionAttributes> sessions = new ArrayList<FeedbackSessionAttributes>();
+        List<FeedbackSessionAttributes> fsInCourse = fsLogic.getFeedbackSessionsForCourse(courseId);
+
+        for (FeedbackSessionAttributes fsa : fsInCourse) {
+            if (!fsa.isPrivateSession() && (fsa.isSentOpenEmail() || fsa.isSentPublishedEmail())) {
+                sessions.add(fsa);
+            }
+        }
+
+        StringBuffer linksFragmentValue = new StringBuffer(1000);
+        String joinUrl = Config.getAppUrl(student.getRegistrationUrl()).toAbsoluteString();
+
+        String joinFragmentValue = isYetToJoinCourse(student)
+                                   ? Templates.populateTemplate(EmailTemplates.FRAGMENT_STUDENT_COURSE_JOIN,
+                                           "${joinUrl}", joinUrl,
+                                           "${courseName}", course.getName())
+                                   : "";
+
+        for (FeedbackSessionAttributes fsa : sessions) {
+
+            String submitUrlHtml = "(Feedback session is " + (fsa.isClosed() ? "closed" : "not yet opened") + ")";
+            String reportUrlHtml = "(Feedback session is not yet published)";
+
+            if (fsa.isPublished()) {
```

Comment:
No, this is incorrect. `fsa.isOpened()` is the correct condition for `submitUrl` to be a valid URL. You do not need to wait for the session to be published to be able to submit, do you?

What needs to be done is:
```java
if (fsa.isOpened()) {
    String submitUrl = ...;
    submitUrlHtml = ...;
}
if (fsa.isPublished()) {
    String reportUrl = ...;
    reportUrlHtml = ...;
}
```